src/core/detail/bitpacking.cc

```diff
@@ -104,6 +108,33 @@ static inline pair<const char*, uint8_t*> simd_variant2_pack(const char* ascii,
 // See https://github.com/lemire/fastvalidate-utf-8/
 // The function returns true (1) if all chars passed in src are
 // 7-bit values (0x00..0x7F). Otherwise, it returns false (0).
+#ifdef __s390x__
+bool validate_ascii_fast(const char* src, size_t len) {
+  size_t i = 0;
+  vector unsigned char has_error = vec_splat_s8(0);
+  if (len >= 16) {
+    for (; i <= len - 16; i += 16) {
+      vector unsigned char current_bytes = vec_load_len((signed char*)(src + i), 16);
+      has_error = vec_orc(has_error, current_bytes);
```

Comment:
similarly what vec_ord does.

btw, why the var is called has_error?

Reply:
The variable 'has_errors' is a 128-bit length vector and it stores the outcome of the OR operations between the vectors(should be at least two OR vectorization operations). 
I reuse the same name to ease the user's comparison between the function variations.